protostar/commands/test/test_results_queue.py

```diff
@@ -1,17 +1,28 @@
+from multiprocessing.managers import ValueProxy
 from typing import TYPE_CHECKING
 
-from protostar.commands.test.test_cases import TestCaseResult
+from protostar.commands.test.test_cases import TestCaseResult, PassedTestCase
 
 if TYPE_CHECKING:
     import queue
 
 
 class TestResultsQueue:
-    def __init__(self, shared_queue: "queue.Queue[TestCaseResult]") -> None:
+    def __init__(
+        self,
+        shared_queue: "queue.Queue[TestCaseResult]",
+        any_failed_or_broken_shared_value: ValueProxy,
+    ) -> None:
         self._shared_queue = shared_queue
+        self._any_failed_or_broken_shared_value = any_failed_or_broken_shared_value
 
     def get(self) -> TestCaseResult:
         return self._shared_queue.get(block=True, timeout=1000)
 
     def put(self, item: TestCaseResult) -> None:
+        if not isinstance(item, PassedTestCase):
+            self._any_failed_or_broken_shared_value.value = True
```

Comment:
I think it is a bit confusing that queue sets a global value. Generally you don't expect that whent you put something on a queue it will change some global flag :p. 

Generally I think we should avoid introducing global state when multiprocessing at all cost, it makes code hard to reason about. 

What do you think about instead of introducing this shared value we would throw an custom error in logger if exit_first flag is set and terminate processes accordingly? 

I know it is weird that logger would handle this as it supposed to log stuff, but I think we will need to gravitate in the future towards making it more of a like results collector with wider responsibilites. You can play around with naming if you want a bit but generally my point is we should keep it in the main process.

Does that make sense?

Reply:
I don't think that flag is particularly controversial, personally I don't find it too different from things like queue.empty(), after all we are only keeping information about it's current state. Changing the logger into something more elaborate might introduce more confusion than a single state variable.

Reply:
It's generally best to avoid shared state when dealing with multiprocessing. In this case it does not much mess as it is only one flag, but it forces you to keep it in mind when you work on this code. What if we increased number of variables (which is very probable to happen in the future)? We are not in rush or anything so I would prefer to do this in a future-proof, stateless way.

I think would be nice to have 
```
class TestResultsCoordinator
    live_logger: TestLiveLogger
    
    def listen():
        handles exit first
        passes to live logger
```
or something similar to avoid confision you mentioned.

Also setting it when you put on the queue is definitely incorrect, this one has to be changed, it is a drity side effect nobody expects.

Reply:
I don't think it is possible to create the TestResultsCoordinator without completely overhauling the TestingLiveLogger as we still need the queue's elements to check whether we found a failed/broken one or not.  I can rewrite the Logger into a Collector (with a separate logger that has nothing to do with the queue or with a logging function) if you want but I just realised that this is what our Queue actually is at the moment - a stateful Collector - perhaps this is something worth embracing, something here has to store the state after all.

Reply:
As we discussed on slack, resolving